lib/player.js

```diff
@@ -5198,6 +5209,19 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     }
   }
 
+  /**
+   * Check variants will apply restrictions on variants in order to reenable
```

Comment:
This JSDoc should probably specify that this is part of the mechanism that disables variants that get HTTP errors. Also, it should probably specify that this might cause a new variant to be chosen.

Something like:
```js
  /**
   * Re-apply restrictions to the variants, to re-enable variants that were
   * temporarily disabled due to network errors.
   * If any variants are enabled this way, a new variant might be chosen for
   * playback.
   * @private
   */
```